ugvc/mrd/srsnv_training_utils.py

```diff
@@ -185,28 +189,23 @@ def prepare_featuremap_for_model(
     if random_seed is not None:
         np.random.seed(random_seed)
     if balanced_sampling_info_fields:
-        # TODO test this code and add a test, I wrote an implementation but didn't even get to run it
-        logger.warning(
-            "Data motif balancing not tested, use at your own risk"
-            f" (balanced_sampling_info_fields={balanced_sampling_info_fields})"
-        )
         # count the number of entries with each combination of info fields to balance by
         balanced_sampling_info_fields_counter = defaultdict(int)
         with pysam.VariantFile(intersect_featuremap_vcf) as fmap:
             for record in fmap.fetch():
                 balanced_sampling_info_fields_counter[
-                    (record.info.get(info_field) for info_field in balanced_sampling_info_fields_counter)
+                    tuple(record.info.get(info_field) for info_field in balanced_sampling_info_fields)
                 ] += 1
         # determine sampling rate per group (defined by a combination of info fields)
         total_size_per_group = total_size // len(balanced_sampling_info_fields_counter)
```

Comment:
I'm confused by this calculation.
Isn't it more natural to count the fraction of events with this info-fields from the total?
And then use the inverse of that to drive the balancing?

Reply:
1. I removed the overhead factor for simplicity
2. total_size_per_group is the total number of the entries we want in the output, and I am dividing it by the total number of groups we are balancing across to get total_size_per_group
3. For each group I am then dividing total_size_per_group by the number of total number of featuremap entries in the group to get the rate at which I want to downsample. I made the variable names more explicit.